imports/plugins/included/product-variant/components/customer/productGridItem.js

```diff
@@ -0,0 +1,191 @@
+
+    if (isSoldOut) {
+      if (isBackorder) {
+        noticeContent.defaultValue = "Backorder";
+        noticeContent.i18nKey = "productDetail.backOrder";
+      } else {
+        noticeContent.defaultValue = "Sold Out!";
+        noticeContent.i18nKey = "productDetail.soldOut";
+      }
+    } else if (isLowQuantity) {
+      noticeContent.defaultValue = "Limited Supply";
+      noticeContent.i18nKey = "productDetail.limitedSupply";
+    }
+
+    return (
+      <div className="grid-alerts">
+        <div className="product-grid-badges">
+          <span className={noticeContent.classNames}>
+            <Components.Translation defaultValue={noticeContent.defaultValue} i18nKey={noticeContent.i18nKey} />
+          </span>
+        </div>
+      </div>
+    );
+  }
+
+  // render product image
+  renderMedia() {
+    const { product } = this.props;
+    const MEDIA_PLACEHOLDER = "/resources/placeholder.gif";
+    const { large } = (Array.isArray(product.media) && product.media[0]) || { large: MEDIA_PLACEHOLDER };
+
+    return (
+      <span
+        className="product-image"
+        style={{ backgroundImage: `url("${large}")` }}
+      />
+    );
+  }
+
+
+  renderAdditionalMedia() {
+    const { product: { media }, position: { weight } } = this.props;
+
+    // if product is not medium weight
+    // or the media array is empty exit
+    if (weight !== 1 || (!media || media.length === 0)) return;
+
+    // creating an additional madia array with
+    // the 2nd, 3rd and 4th images returned
+    // in the media array
+    const additionalMedia = [...media.slice(1, 4)];
+
+    return (
+      <div className="product-additional-images">
+        {additionalMedia.map((img) => (
+          <span
+            key={img.small}
+            className="product-image"
+            style={{ backgroundImage: `url("${img.medium}")` }}
+          />
+        ))}
+      </div>
+    );
+  }
+
+  renderGridContent() {
+    const { product } = this.props;
+
+    return (
+      <div className="grid-content">
+        <a
+          href={this.productURL}
+          data-event-category="grid"
+          data-event-action="product-click"
+          data-event-label="grid product click"
+          data-event-value={product._id}
+          onClick={this.handleClick}
+        >
+          <div className="overlay">
+            <div className="overlay-title">{product.title}</div>
+            <div className="currency-symbol">{formatPriceString(product.price.range)}</div>
+            {this.props.isSearch &&
+                <div className="overlay-description">{product.description}</div>
+            }
+          </div>
+        </a>
+      </div>
+    );
+  }
+
+  render() {
+    const { product, isSearch } = this.props;
+    return (
+      <li
+        className={this.productClassNames}
+        data-id={product._id}
+        id={product._id}
+      >
+        <div className={(isSearch) ? "item-content" : ""}>
```

Comment:
Why do we allow ternary's in React components, but nowhere else?

More of a discussion overall, since the code here is OK based on our docs.
https://docs.reactioncommerce.com/reaction-docs/master/styleguide#methods